__tests__/app.test.js

```diff
@@ -271,23 +271,25 @@ describe("GET /api/articles", () => {
           expect(allArticles).toHaveLength(12);
         });
     });
-    test("endpoint responds with an array of article objects, which include specified keys", () => {
+    test("endpoint responds with an array of article objects, which include specified keys and correct data types", () => {
       return request(app)
         .get("/api/articles")
         .expect(200)
         .then(({ body }) => {
           const { allArticles } = body;
           allArticles.forEach((article) => {
-            expect(article.hasOwnProperty("username")).toBe(true);
-            expect(article.hasOwnProperty("title")).toBe(true);
-            expect(article.hasOwnProperty("article_id")).toBe(true);
-            expect(article.hasOwnProperty("topic")).toBe(true);
-            expect(article.hasOwnProperty("created_at")).toBe(true);
-            expect(article.hasOwnProperty("votes")).toBe(true);
-            expect(article.hasOwnProperty("comment_count")).toBe(true);
+            expect(article.hasOwnProperty("author", expect.any(String))).toBe(true);
+            expect(article.hasOwnProperty("title", expect.any(String))).toBe(true);
+            expect(article.hasOwnProperty("article_id", expect.any(Number))).toBe(true);
+            expect(article.hasOwnProperty("topic", expect.any(String))).toBe(true);
+            expect(article.hasOwnProperty("created_at", expect.any(String))).toBe(true);
+            expect(article.hasOwnProperty("votes", expect.any(Number))).toBe(true);
+            expect(article.hasOwnProperty("comment_count", expect.any(Number))).toBe(true);
           });
```

Comment:
Sorry Sol, I misread this when we spoke last night... you can only do this with a specific jest matcher like so... ` expect(article).toHaveProperty('author', expect.any(String));`